bin/internal/update_dart_sdk.ps1

```diff
@@ -20,9 +20,16 @@ $cachePath = "$flutterRoot\bin\cache"
 $dartSdkPath = "$cachePath\dart-sdk"
 $dartSdkLicense = "$cachePath\LICENSE.dart_sdk_archive.md"
 $engineStamp = "$cachePath\engine-dart-sdk.stamp"
-$engineVersion = (Get-Content "$flutterRoot\bin\internal\engine.version")
 $engineRealm = (Get-Content "$flutterRoot\bin\internal\engine.realm")
 
+if (Test-Path "$flutterRoot\bin\internal\engine.version") {
+    $engineVersion = (Get-Content "$flutterRoot\bin\internal\engine.version")
+} else {
+    # Calculate the engine hash from tracked git files.
+    $lsTree = ((git ls-tree -r HEAD engine DEPS) | Out-String).Replace("`r`n", "`n")
```

Comment:
I was gonna ask why powershell uses backticks for escapes in strings, but I'm guessing it's because then you would need to escape all backslashes in paths... *sigh*

Reply:
I asked the same question but moved along instead of trying to reason with Windows :)